shell/platform/android/test/io/flutter/embedding/engine/systemchannels/ScribeChannelTest.java

```diff
@@ -0,0 +1,104 @@
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.StandardMethodCodec;
+import java.nio.ByteBuffer;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.robolectric.annotation.Config;
+
+@RunWith(AndroidJUnit4.class)
+public class ScribeChannelTest {
+  private static BinaryMessenger.BinaryReply sendToBinaryMessageHandler(
+      BinaryMessenger.BinaryMessageHandler binaryMessageHandler, String method) {
+    MethodCall methodCall = new MethodCall(method, null);
+    ByteBuffer encodedMethodCall = StandardMethodCodec.INSTANCE.encodeMethodCall(methodCall);
+    BinaryMessenger.BinaryReply mockReply = mock(BinaryMessenger.BinaryReply.class);
+    binaryMessageHandler.onMessage((ByteBuffer) encodedMethodCall.flip(), mockReply);
+    return mockReply;
+  }
+
+  ScribeChannel.ScribeMethodHandler mockHandler;
+  BinaryMessenger.BinaryMessageHandler binaryMessageHandler;
+
+  @Before
+  public void setUp() {
+    ArgumentCaptor<BinaryMessenger.BinaryMessageHandler> binaryMessageHandlerCaptor =
+        ArgumentCaptor.forClass(BinaryMessenger.BinaryMessageHandler.class);
+    DartExecutor mockBinaryMessenger = mock(DartExecutor.class);
+    mockHandler = mock(ScribeChannel.ScribeMethodHandler.class);
+    ScribeChannel scribeChannel = new ScribeChannel(mockBinaryMessenger);
+
+    scribeChannel.setScribeMethodHandler(mockHandler);
+
+    verify((BinaryMessenger) mockBinaryMessenger, times(1))
+        .setMessageHandler(any(String.class), binaryMessageHandlerCaptor.capture());
+
+    binaryMessageHandler = binaryMessageHandlerCaptor.getValue();
+  }
+
+  @Config(minSdk = API_LEVELS.API_34)
+  @TargetApi(API_LEVELS.API_34)
+  @Test
+  public void respondsToStartStylusHandwriting() {
+    BinaryMessenger.BinaryReply mockReply =
+        sendToBinaryMessageHandler(
+            binaryMessageHandler, ScribeChannel.METHOD_START_STYLUS_HANDWRITING);
+
+    verify(mockReply).reply(any(ByteBuffer.class));
+    verify(mockHandler).startStylusHandwriting();
+  }
+
+  @Config(minSdk = API_LEVELS.API_34)
+  @TargetApi(API_LEVELS.API_34)
+  @Test
+  public void respondsToIsStylusHandwritingAvailable() {
+    BinaryMessenger.BinaryReply mockReply =
+        sendToBinaryMessageHandler(
+            binaryMessageHandler, ScribeChannel.METHOD_IS_STYLUS_HANDWRITING_AVAILABLE);
+
+    verify(mockReply).reply(any(ByteBuffer.class));
+    verify(mockHandler).isStylusHandwritingAvailable();
+  }
+
+  @Config(sdk = API_LEVELS.API_32)
+  @TargetApi(API_LEVELS.API_32)
+  @Test
+  public void respondsToStartStylusHandwritingWhenAPILevelUnsupported() {
+    BinaryMessenger.BinaryReply mockReply =
+        sendToBinaryMessageHandler(
+            binaryMessageHandler, ScribeChannel.METHOD_START_STYLUS_HANDWRITING);
+
+    verify(mockReply).reply(any(ByteBuffer.class));
```

Comment:
I would have liked to test the exact argument that `reply` was called with here, but I was struggling to do anything useful with the ByteBuffer. I was trying `StandardMethodCodec.INSTANCE.decodeEnvelope` because it looks like that's what it's encoded with (e.g. [here](https://github.com/flutter/engine/blob/fd6082ab669356ae1525adc809cc0d0c78109b63/shell/platform/android/io/flutter/plugin/common/MethodChannel.java#L277)), but I always got an underflow error.

Reply:
Instead of trying to reverse engineer the reply bytebuffer what if you used a spy on the result callback and verified that either callback.error or result.error was called. 

I do not love that the error condition verifies the same thing as the success condition but at least we verify we do not call the methods we are trying to protect.

Reply:
I changed this to JSONMethodCodec which I believe is what we're supposed to be using for these kinds of method channels now, and incidentally I was able to decode and test the value of `result` then.